src/openrct2/object/RideObject.cpp

```diff
@@ -434,7 +434,11 @@ void RideObject::ReadLegacyVehicle(
     vehicle->effect_visual = stream->ReadValue<uint8_t>();
     vehicle->draw_order = stream->ReadValue<uint8_t>();
     vehicle->num_vertical_frames_override = stream->ReadValue<uint8_t>();
-    stream->Seek(4, STREAM_SEEK_CURRENT);
+    stream->Seek(1, STREAM_SEEK_CURRENT);
+    vehicle->animation_speed_modifier = stream->ReadValue<uint8_t>();
+    vehicle->steam_effect_modifier[0] = stream->ReadValue<uint8_t>();
+    vehicle->steam_effect_modifier[1] = stream->ReadValue<uint8_t>();
+//    stream->Seek(0, STREAM_SEEK_CURRENT);
```

Comment:
New fields should only be added to the JSON format so this needs to be reverted.